Fix #23: Uniform exception handling across all OpenFAM APIs

faizan-barmawer opened the pull request:
This fix includes following changes:
- All APIs return Fam_Exception on failure
- Change in API specification where return value is not effectively used.
  Return value changed from int to void for fam_initialize, etc.
- Error message includes function name and line no., from where the exception
  is raised.

mashoodk commented on include/fam/fam.h:
```diff
@@ -288,7 +288,7 @@ class fam {
      * @see #Fam_Options
      */
-    int fam_initialize(const char *groupName, Fam_Options *options);
+    void fam_initialize(const char *groupName, Fam_Options *options);
```
Please change description

faizan-barmawer commented:
Addressed review comments.
Pushed a new commit with 
- changes to API description in fam.h header file.
- Format message on memoryserver exceptions using ERROR_MSG macro.

faizan-barmawer commented on include/fam/fam.h:
```diff
@@ -436,8 +433,8 @@ class fam {
      */
-    int fam_change_permissions(Fam_Region_Descriptor *descriptor,
-                               mode_t accessPermissions);
+    void fam_change_permissions(Fam_Region_Descriptor *descriptor,
```
Missed to update the description for fam_change_permission.
Also verified description of other API which returns void. They look fine.

Regards,
Faizan

faizan-barmawer commented:
Updated this PR, resolving review comments.

This PR was showing the below merged commit, which I did a rebased/resolved merged conflicts as part of the update.
   - Multiple memoryserver implementation phase 1 (#55) 

As  mentioned in some git blogs, the solution to this problem is to perform the following steps:
 - checkout the working/feature branch which is to be merged with target (master/devel) branch
    git checkout <your working/feature branch>
 - Fetch the latest changes on the target branch 
    git fetch upstream
  - Rebase your feature/working branch to the target merge branch.
    git rebase upstream/devel
  -  Force push your repository/working branch
    git push --force origin

faizan-barmawer commented on include/fam/fam.h:
```diff
@@ -273,29 +273,32 @@ class fam {
      * @see #Fam_Options
      */
-    int fam_initialize(const char *groupName, Fam_Options *options);
+    void fam_initialize(const char *groupName, Fam_Options *options);
```
Added a note in fam.h and pushed a new commit.
